src/super_gradients/training/models/detection_models/customizable_detector.py

```diff
@@ -96,3 +101,30 @@ def replace_head(self, new_num_classes: Optional[int] = None, new_head: Optional
             self.heads_params = factory.insert_module_param(self.heads_params, "num_classes", new_num_classes)
             self.heads = factory.get(factory.insert_module_param(self.heads_params, "in_channels", self.neck.out_channels))
             self._initialize_weights(self.bn_eps, self.bn_momentum, self.inplace_act)
+
+    @staticmethod
+    def get_post_prediction_callback(conf: float, iou: float) -> DetectionPostPredictionCallback:
+        raise NotImplementedError
+
+    def set_dataset_processing_params(self, class_names: Optional[List[str]], image_processor: Optional[Processing]) -> None:
+        """Set the processing parameters for the dataset.
+
+        :param class_names:     (Optional) Names of the dataset the model was trained on.
+        :param image_processor: (Optional) Image processing objects to reproduce the dataset preprocessing used for training.
+        """
+        self._class_names = class_names or self._class_names
+        self._image_processor = image_processor or self._image_processor
+
+    def predict(self, images, iou: float = 0.65, conf: float = 0.01) -> DetectionResults:
```

Comment:
Continuing my previous comment - predict is here to visualize results ad not to optimize mAP, hence the values should be selected accordingly.
This means that the default values should be related to the dataset on which the model was trained on.
It also looks way better IMO if all the users needs to do is call predict on their images.

Reply:
Done here d974313215196cc3c4210f1c49682040b93285cf